shell/platform/embedder/embedder.h

```diff
@@ -1205,6 +1209,19 @@ typedef struct {
   ///
   /// Embedders can provide either snapshot buffers or aot_data, but not both.
   FlutterEngineAOTData aot_data;
+
+  /// A callback that computes the locale the platform would natively resolve
+  /// to.
+  ///
+  /// The input parameter is an array of null terminated char* strings. Each
+  /// group of 3 strings represents one locale, with the strings supplying
+  /// language, country, and script code in that order. Language code cannot be
+  /// empty, but the other two may be null.
```

Comment:
So it's not really a null-terminated array; it's an array that ends if there's a null at an index divisible by three. Fixing the above would resolve this oddity as well.